ewatercycle/forcing/__init__.py

```diff
@@ -39,8 +39,9 @@ def generate(target_model: str,
     if constructor is None:
         raise NotImplementedError(f'Target model `{target_model}` is not supported by the eWatercycle forcing generator')
     if model_specific_options is None:
-        return constructor.generate(dataset, start_time, end_time, shape)
-    forcing_info = constructor.generate(dataset, start_time, end_time, shape, **model_specific_options)
+        forcing_info = constructor.generate(dataset, start_time, end_time, shape)
+    else:
+        forcing_info = constructor.generate(dataset, start_time, end_time, shape, **model_specific_options)
```

Comment:
Can we get rid of this conditional block if we make `model_specific_options` default to an empty dict instead of `None`?

Reply:
Empty dict is mutable, which is not allowed as a default value for a argument. As it makes the value tied to the class instead of the object. Can make it simpler